### PR TITLE
[ACA-100] View a previous version

### PR DESCRIPTION
**Pull request by ursedaniel**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://issues.alfresco.com/jira/browse/ACA-100
I can only download a version for a node, I'm not able to view it's rendition.

**What is the new behaviour?**
Now I can view the rendition/content of a previous version.

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/core/viewer/components/viewer.component.ts [hunk 1]
```diff
@@ -720,4 +768,86 @@ export class ViewerComponent implements OnChanges, OnInit, OnDestroy {
...
+                }
+
+                this.urlFileContent = this.apiService.contentApi.getVersionRenditionUrl(nodeId, versionId, renditionId);
+            }
+        } catch (err) {
+            this.logService.error(err);
+        }
+    }
+
+    private async resolveVersionRendition(nodeId: string, versionId: string, renditionId: string): Promise<RenditionEntry> {
```
Comment (eromano) [inline lib/core/viewer/components/viewer.component.ts:791]: this code seems quite duplicated of the normal version retention I think we should try to unify the common logic and move it in a service level

File: docs/content-services/components/version-manager.component.md [hunk 2]
```diff
@@ -42,6 +42,7 @@ Displays the version history of a node with the ability to upload a new version.
 | uploadCancel | [`EventEmitter`](https://angular.io/api/core/EventEmitter)`<boolean>` | Emitted when an cancelling during upload. |
 | uploadError | [`EventEmitter`](https://angular.io/api/core/EventEmitter)`<`[`Node`](https://github.com/Alfresco/alfresco-js-api/blob/development/src/api/content-rest-api/docs/Node.md)`>` | Emitted when an error occurs during upload. |
 | uploadSuccess | [`EventEmitter`](https://angular.io/api/core/EventEmitter)`<`[`Node`](https://github.com/Alfresco/alfresco-js-api/blob/development/src/api/content-rest-api/docs/Node.md)`>` | Emitted when a file is uploaded successfully. |
+| viewVersion | [`EventEmitter`](https://angular.io/api/core/EventEmitter)`<string>` | Emitted when viewing a version. |
```
Comment (eromano) [inline docs/content-services/components/version-manager.component.md:45]: we should add a notation that identifies that this endpoint is available only in 7

File: docs/content-services/components/version-list.component.md [hunk 3]
```diff
@@ -32,6 +33,7 @@ Displays the version history of a node in a [Version Manager component](version-
 | ---- | ---- | ----------- |
 | deleted | [`EventEmitter`](https://angular.io/api/core/EventEmitter)`<`[`Node`](https://github.com/Alfresco/alfresco-js-api/blob/development/src/api/content-rest-api/docs/Node.md)`>` | Emitted when a version is deleted |
 | restored | [`EventEmitter`](https://angular.io/api/core/EventEmitter)`<`[`Node`](https://github.com/Alfresco/alfresco-js-api/blob/development/src/api/content-rest-api/docs/Node.md)`>` | Emitted when a version is restored |
+| viewVersion | [`EventEmitter`](https://angular.io/api/core/EventEmitter)`<string>` | Emitted when viewing a version |
```
Comment (eromano) [inline docs/content-services/components/version-list.component.md:36]: we should add a notation that identifies that this endpoint is available only in 7

File: lib/core/directives/node-download.directive.spec.ts [hunk 4]
```diff
@@ -26,10 +26,11 @@ import { CoreTestingModule } from '../testing/core.testing.module';
 import { TranslateModule } from '@ngx-translate/core';
 
 @Component({
-    template: '<div [adfNodeDownload]="selection"></div>'
+    template: '<div [adfNodeDownload]="selection" [version]="version"></div>'
```
Comment (eromano) [inline lib/core/directives/node-download.directive.spec.ts:29]: this needs a documentation update

File: lib/core/viewer/components/viewer.component.ts [hunk 5]
```diff
@@ -32,16 +32,25 @@ import { Subscription } from 'rxjs';
...
     selector: 'adf-viewer',
     templateUrl: './viewer.component.html',
     styleUrls: ['./viewer.component.scss'],
-    host: { 'class': 'adf-viewer' },
+    host: {'class': 'adf-viewer'},
     encapsulation: ViewEncapsulation.None
 })
 export class ViewerComponent implements OnChanges, OnInit, OnDestroy {
 
+    constructor(private apiService: AlfrescoApiService,
```
Comment (eromano) [inline lib/core/viewer/components/viewer.component.ts:47]: move back the contractor where it was has to be after the properties

File: lib/core/viewer/components/viewer.component.ts [hunk 6]
```diff
@@ -672,31 +698,31 @@ export class ViewerComponent implements OnChanges, OnInit, OnDestroy {
         return rendition;
     }
 
-    private async waitRendition(nodeId: string, renditionId: string): Promise<RenditionEntry> {
```
Comment (eromano) [inline lib/core/viewer/components/viewer.component.ts:675]: all the logic related of the rendition can be moved in a service of this component ?

File: lib/core/viewer/components/viewer.component.ts (hunk 5 above)
Comment (ursedaniel) [reply lib/core/viewer/components/viewer.component.ts:47]: auto align did it, didnt even realised lol

File: lib/core/directives/node-download.directive.spec.ts (hunk 4 above)
Comment (ursedaniel) [reply lib/core/directives/node-download.directive.spec.ts:29]: https://github.com/Alfresco/alfresco-ng2-components/pull/5785/files#diff-1b2eada0dcb743cd0b35f51264138b3c

**Comment by DenysVuika** [comment]
You will have troubles rebasing this PR next time because you merged develop here

**Comment by KikoUK** [comment]
> You will have troubles rebasing this PR next time because you merged develop here

Thanks, reverted the merge.

**Comment by ursedaniel** [comment]
created a new PR fo quickly fix all the conflicts of the rebase https://github.com/Alfresco/alfresco-ng2-components/pull/5828

**Pull request closed without merging**